src/loaders/jobs/taskfarmer/task_generator.py

```diff
@@ -292,6 +306,7 @@ def main():
     root_dir = args.root_dir
 
     task_mgr = TFTaskManager(kbase_collection, load_ver, tool, source_data_dir, root_dir=root_dir)
+    task_mgr.check_preconditions(args.force)
```

Comment:
I think we should just do the precondition check before writing batch scripts. When the previous job is in complete/running/pending status, we shouldn't update batch scripts. However, when '--force' or the previous job is in failed/canceled/timeout status, we should update the batch script and allow users to resubmit jobs with the same/different arguments.

Reply:
after precondition checking is passed, (the previous job is in failed/canceled/timeout status or --force is True), it's safe to overwrite existing batch files.

Reply:
:+1: